Settings.c

```diff
@@ -826,6 +826,12 @@ Settings* Settings_new(unsigned int initialCpuCount, Hashtable* dynamicMeters, H
          legacyDotfile = NULL;
       }
    }
+   char* resolvedFilename = xMalloc(PATH_MAX);
+   if (realpath(this->filename, resolvedFilename)) {
+       free(this->filename);
+       this->filename = resolvedFilename;
+   }
```

Comment:
Memory leak for memory returned by `xMalloc` when `realpath` fails.

Also, cf. `free_and_xStrdup` …